### PR TITLE
Rename csi-driver to better short-name.

### PR DESCRIPTION
**Pull request by srust**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

File: templates/flavors/default/kustomization.yaml [hunk 1]
```diff
@@ -29,6 +30,7 @@ patches:
         labels:
             cni: cilium
             ccm: linode
+            csi: linode
```
Comment (AshleyDumaine) [inline templates/flavors/default/kustomization.yaml:33]: If we want to have the linode csi driver installed by default, can we add a note to the docs that this is the case?
```
~~~admonish success title=""
Installed by default
~~~
```

File: config/crd/bases/infrastructure.cluster.x-k8s.io_linodeclustertemplates.yaml [hunk 2]
```diff
@@ -3,7 +3,7 @@ apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.14.0
+    controller-gen.kubebuilder.io/version: v0.13.0
```
Comment (AshleyDumaine) [inline config/crd/bases/infrastructure.cluster.x-k8s.io_linodeclustertemplates.yaml:6]: did devbox change the version here again or is this an intentional downgrade?

File: templates/flavors/default/kustomization.yaml (hunk 1 above)
Comment (srust) [reply templates/flavors/default/kustomization.yaml:33]: Sure

File: config/crd/bases/infrastructure.cluster.x-k8s.io_linodeclustertemplates.yaml (hunk 2 above)
Comment (srust) [reply config/crd/bases/infrastructure.cluster.x-k8s.io_linodeclustertemplates.yaml:6]: devbox did not - Makefile non-devbox did apparently. Depends where things are generated from devbox or not... @eljohnson92 HALP!

**Review by AshleyDumaine — APPROVED** [review]
LGTM

**Review by eljohnson92 — APPROVED** [review]

**Pull request merged**